### PR TITLE
[RHPAM-4111]-Certify Spring Boot 2.4.x

### PR DESCRIPTION
**Pull request by VaniHaripriya**
JIRA: https://issues.redhat.com/browse/RHPAM-4111
Upgraded version of Spring Boot dependencies to align with Red Hat supported Spring Boot 2.4.9 version.
This PR should be merged along with a related PR- https://github.com/kiegroup/droolsjbpm-integration/pull/2680

### CODE REVIEW

**Review by fjtirado — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -264,8 +264,8 @@
     <version.org.sonatype.sisu>2.3.0</version.org.sonatype.sisu>
     <version.org.sonatype.sisu.sisu-guice>3.2.3</version.org.sonatype.sisu.sisu-guice>
     <version.org.springframework>5.2.9.RELEASE</version.org.springframework>
-    <version.org.springframework.boot>2.3.4.RELEASE</version.org.springframework.boot>
-    <version.org.springframework.amqpboot>2.3.4</version.org.springframework.amqpboot>
+    <version.org.springframework.boot>2.4.9</version.org.springframework.boot>
+    <version.org.springframework.amqpboot>2.3.6.redhat-00004</version.org.springframework.amqpboot>
```
Comment (gmunozfe) [inline pom.xml:268]: Not sure where is this used, and if it's the right version (not 2.4.x)

**Review by gmunozfe — COMMENTED** [review]
One question and also a request to update droolsjbpm-integration repo (if you haven't done) at

https://github.com/kiegroup/droolsjbpm-integration/blob/c37ce663c39b95e707daf9f9f32be9c3e665d3b9/kie-spring-boot/pom.xml#L36

to point to the right starter for narayana:

According to [this](https://download.eng.brq.redhat.com/released/JBoss-middleware/rhoar/spring-boot/2.4.9/extras/repository-artifact-list.txt), it should be 

**2.6.1.redhat-00001**

File: pom.xml (hunk 1 above)
Comment (VaniHaripriya) [reply pom.xml]: Removed amqp-10-jms-spring-boot-starter as I didn't find any references in kie projects.

**Comment by gmunozfe** [comment]
jenkins run fdb

File: pom.xml (hunk 1 above)
Comment (mareknovotny) [reply pom.xml]: do not use here any version with -redhat-#### suffix

File: pom.xml [hunk 2]
```diff
@@ -264,8 +264,7 @@
     <version.org.sonatype.sisu>2.3.0</version.org.sonatype.sisu>
     <version.org.sonatype.sisu.sisu-guice>3.2.3</version.org.sonatype.sisu.sisu-guice>
     <version.org.springframework>5.2.9.RELEASE</version.org.springframework>
-    <version.org.springframework.boot>2.3.4.RELEASE</version.org.springframework.boot>
-    <version.org.springframework.amqpboot>2.3.4</version.org.springframework.amqpboot>
+    <version.org.springframework.boot>2.4.9</version.org.springframework.boot>
```
Comment (mareknovotny) [inline pom.xml:267]: Shouldn't we use 2.5.x or 2.6.x?

Comment (mareknovotny) [reply pom.xml:267]: ok, i can see the jira now is about to certify SB 2.4.x. so ignore my previous comment

**Review by mareknovotny — APPROVED** [review]

**Comment by mareknovotny** [comment]
jenkins do fdb

**Comment by mareknovotny** [comment]
I guess we need to show in description that https://github.com/kiegroup/droolsjbpm-integration/pull/2680 is related and should not be merged without this PR too, am I right? @VaniHaripriya @gmunozfe

**Comment by mareknovotny** [comment]
there are like 21 failed tests

-   org.jbpm.runtime.manager.impl.KjarRuntimeEnvironmentTest.testScriptTaskFromClasspathContainerNamedKbaseKsession

- org.jbpm.runtime.manager.impl.KjarRuntimeEnvironmentTest.testScriptTaskFromKjarByNameNamedKbaseKsession
- org.jbpm.runtime.manager.impl.KjarRuntimeEnvironmentTest.testScriptTaskFromKjar
- org.jbpm.runtime.manager.impl.KjarRuntimeEnvironmentTest.testXStreamUnmarshalCustomObject
- org.jbpm.runtime.manager.impl.KjarRuntimeEnvironmentTest.testUserTaskFromKjarPPI
- org.jbpm.runtime.manager.impl.KjarRuntimeEnvironmentTest.testUserTaskFromKjar
- org.jbpm.runtime.manager.impl.KjarRuntimeEnvironmentTest.testScriptTaskFromClasspathContainer
- org.jbpm.runtime.manager.impl.KjarRuntimeEnvironmentTest.testScriptTaskFromKjarUsingNamedKbaseKsession
- org.jbpm.runtime.manager.impl.KjarRuntimeEnvironmentTest.testCustomTaskFromKjar
- org.jbpm.runtime.manager.impl.KjarRuntimeEnvironmentTest.testScriptTaskFromKjarByName
- org.jbpm.kie.services.test.KModuleDeploymentServiceNoMavenTest.testDeploymentOfProcesses
- org.jbpm.kie.services.test.KModuleDeploymentServiceNoMavenTest.testDeploymentOfProcessesCustomId
- org.jbpm.process.workitem.bpmn2.BusinessRuleTaskHandlerTest.testDmnBusinessRuleTaskNoScanner
- org.jbpm.process.workitem.bpmn2.BusinessRuleTaskHandlerTest.testDrlStatefulBusinessRuleTaskNoScanner
- org.jbpm.process.workitem.bpmn2.BusinessRuleTaskHandlerTest.testDrlStatelessBusinessRuleTaskNoScanner
- org.jbpm.process.workitem.bpmn2.BusinessRuleTaskTest.testCallingDecisionService
- org.jbpm.process.workitem.bpmn2.BusinessRuleTaskTest.testDecisionTaskProcess
- org.jbpm.process.workitem.bpmn2.BusinessRuleTaskTest.testDecisionPassingNonNullToDMN
- org.jbpm.process.workitem.bpmn2.BusinessRuleTaskTest.testDecisionPassingNullToDMN
- org.jbpm.process.workitem.bpmn2.BusinessRuleTaskTest.testBusinessRuleTaskProcess
- org.jbpm.process.workitem.bpmn2.JaxWSServiceTaskTest.testAsyncServiceInvocation

Please check them and fix if possible

**Comment by gmunozfe** [comment]
> I guess we need to show in description that [kiegroup/droolsjbpm-integration#2680](https://github.com/kiegroup/droolsjbpm-integration/pull/2680) is related and should not be merged without this PR too, am I right? @VaniHaripriya @gmunozfe

yes, both are related to be aligned to ROAR

**Comment by mareknovotny** [comment]
fyi there is also this https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1866 automated PR for Spring framework upgrade. So if you will do this sync here i need to close it. Let me know

**Comment by VaniHaripriya** [comment]
> fyi there is also this #1866 automated PR for Spring framework upgrade. So if you will do this sync here i need to close it. Let me know

@mareknovotny Sure,I will do the Spring framework upgrade with this PR, you may close the other one. Thank you.

**Comment by VaniHaripriya** [comment]
> > I guess we need to show in description that [kiegroup/droolsjbpm-integration#2680](https://github.com/kiegroup/droolsjbpm-integration/pull/2680) is related and should not be merged without this PR too, am I right? @VaniHaripriya @gmunozfe
> 
> yes, both are related for being aligned to ROAR @VaniHaripriya could you update the description with this reference to the other PR following the template? ^

Thank you. The description has been updated with the same.

**Comment by mareknovotny** [comment]
jenkins do fdb

**Comment by mareknovotny** [comment]
FDB shows the enforcer error too like the subsequent PR in droolsjbpm-integration repository 
```
Found in:
[2022-01-17T17:03:29.040Z]     org.apache.geronimo.specs:geronimo-jms_2.0_spec:jar:1.0-alpha-2:compile
[2022-01-17T17:03:29.040Z]     jakarta.jms:jakarta.jms-api:jar:2.0.3:compile
[2022-01-17T17:03:29.040Z]   Duplicate classes:
[2022-01-17T17:03:29.040Z]     javax/jms/QueueConnection.class
[2022-01-17T17:03:29.040Z]     javax/jms/MessageNotReadableException.class
[2022-01-17T17:03:29.040Z]     javax/jms/XAJMSContext.class
[2022-01-17T17:03:29.040Z]     javax/jms/JMSDestinationDefinitions.class
[2022-01-17T17:03:29.040Z]     javax/jms/InvalidDestinationRuntimeException.class
[2022-01-17T17:03:29.040Z]     javax/jms/ObjectMessage.class
[2022-01-17T17:03:29.040Z]     javax/jms/TopicRequestor.class
[2022-01-17T17:03:29.040Z]     javax/jms/JMSException.class
[2022-01-17T17:03:29.040Z]     javax/jms/InvalidSelectorException.class
[2022-01-17T17:03:29.040Z]     javax/jms/Message.class
[2022-01-17T17:03:29.040Z]     javax/jms/JMSConnectionFactoryDefinition.class
[2022-01-17T17:03:29.040Z]     javax/jms/JMSSecurityException.class
[2022-01-17T17:03:29.040Z]     javax/jms/IllegalStateRuntimeException.class
[2022-01-17T17:03:29.040Z]     javax/jms/ServerSessionPool.class
[2022-01-17T17:03:29.040Z]     javax/jms/Topic.class
[2022-01-17T17:03:29.040Z]     javax/jms/ConnectionFactory.class
[2022-01-17T17:03:29.040Z]     javax/jms/ConnectionMetaData.class
[2022-01-17T17:03:29.040Z]     javax/jms/MessageListener.class
[2022-01-17T17:03:29.040Z]     javax/jms/XAConnectionFactory.class
[2022-01-17T17:03:29.040Z]     javax/jms/Destination.class
[2022-01-17T17:03:29.040Z]     javax/jms/JMSConnectionFactoryDefinitions.class
[2022-01-17T17:03:29.040Z]     javax/jms/MessageConsumer.class
[2022-01-17T17:03:29.040Z]     javax/jms/QueueRequestor.class
[2022-01-17T17:03:29.040Z]     javax/jms/QueueBrowser.class
[2022-01-17T17:03:29.040Z]     javax/jms/CompletionListener.class
[2022-01-17T17:03:29.040Z]     javax/jms/XASession.class
[2022-01-17T17:03:29.040Z]     javax/jms/IllegalStateException.class
[2022-01-17T17:03:29.040Z]     javax/jms/TransactionInProgressException.class
[2022-01-17T17:03:29.040Z]     javax/jms/MessageFormatRuntimeException.class
[2022-01-17T17:03:29.040Z]     javax/jms/JMSContext.class
[2022-01-17T17:03:29.040Z]     javax/jms/InvalidSelectorRuntimeException.class
[2022-01-17T17:03:29.040Z]     javax/jms/ResourceAllocationException.class
[2022-01-17T17:03:29.040Z]     javax/jms/XATopicConnectionFactory.class
[2022-01-17T17:03:29.040Z]     javax/jms/JMSConnectionFactory.class
[2022-01-17T17:03:29.040Z]     javax/jms/JMSPasswordCredential.class
[2022-01-17T17:03:29.040Z]     javax/jms/StreamMessage.class
[2022-01-17T17:03:29.040Z]     javax/jms/Queue.class
[2022-01-17T17:03:29.040Z]     javax/jms/ResourceAllocationRuntimeException.class
[2022-01-17T17:03:29.040Z]     javax/jms/JMSDestinationDefinition.class
[2022-01-17T17:03:29.040Z]     javax/jms/MessageNotWriteableException.class
[2022-01-17T17:03:29.040Z]     javax/jms/DeliveryMode.class
[2022-01-17T17:03:29.040Z]     javax/jms/TopicConnection.class
[2022-01-17T17:03:29.040Z]     javax/jms/MessageFormatException.class
[2022-01-17T17:03:29.040Z]     javax/jms/XAQueueConnection.class
[2022-01-17T17:03:29.040Z]     javax/jms/XAConnection.class
[2022-01-17T17:03:29.040Z]     javax/jms/MessageProducer.class
[2022-01-17T17:03:29.040Z]     javax/jms/TemporaryQueue.class
[2022-01-17T17:03:29.040Z]     javax/jms/TopicConnectionFactory.class
[2022-01-17T17:03:29.040Z]     javax/jms/XAQueueSession.class
[2022-01-17T17:03:29.040Z]     javax/jms/InvalidClientIDException.class
[2022-01-17T17:03:29.040Z]     javax/jms/BytesMessage.class
[2022-01-17T17:03:29.040Z]     javax/jms/QueueSender.class
[2022-01-17T17:03:29.040Z]     javax/jms/TransactionRolledBackException.class
[2022-01-17T17:03:29.040Z]     javax/jms/JMSSessionMode.class
[2022-01-17T17:03:29.040Z]     javax/jms/Connection.class
[2022-01-17T17:03:29.040Z]     javax/jms/MessageEOFException.class
[2022-01-17T17:03:29.040Z]     javax/jms/JMSRuntimeException.class
[2022-01-17T17:03:29.040Z]     javax/jms/ExceptionListener.class
[2022-01-17T17:03:29.040Z]     javax/jms/ConnectionConsumer.class
[2022-01-17T17:03:29.040Z]     javax/jms/ServerSession.class
[2022-01-17T17:03:29.040Z]     javax/jms/JMSSecurityRuntimeException.class
[2022-01-17T17:03:29.040Z]     javax/jms/MessageNotWriteableRuntimeException.class
[2022-01-17T17:03:29.040Z]     javax/jms/QueueSession.class
[2022-01-17T17:03:29.040Z]     javax/jms/TemporaryTopic.class
[2022-01-17T17:03:29.040Z]     javax/jms/TopicPublisher.class
[2022-01-17T17:03:29.040Z]     javax/jms/XAQueueConnectionFactory.class
[2022-01-17T17:03:29.040Z]     javax/jms/QueueReceiver.class
[2022-01-17T17:03:29.040Z]     javax/jms/MapMessage.class
[2022-01-17T17:03:29.040Z]     javax/jms/InvalidClientIDRuntimeException.class
[2022-01-17T17:03:29.040Z]     javax/jms/TransactionRolledBackRuntimeException.class
[2022-01-17T17:03:29.040Z]     javax/jms/Session.class
[2022-01-17T17:03:29.040Z]     javax/jms/TextMessage.class
[2022-01-17T17:03:29.040Z]     javax/jms/TopicSession.class
[2022-01-17T17:03:29.040Z]     javax/jms/TopicSubscriber.class
[2022-01-17T17:03:29.040Z]     javax/jms/InvalidDestinationException.class
[2022-01-17T17:03:29.040Z]     javax/jms/TransactionInProgressRuntimeException.class
[2022-01-17T17:03:29.040Z]     javax/jms/JMSProducer.class
[2022-01-17T17:03:29.040Z]     javax/jms/JMSConsumer.class
[2022-01-17T17:03:29.040Z]     javax/jms/QueueConnectionFactory.class
[2022-01-17T17:03:29.040Z]     javax/jms/XATopicConnection.class
[2022-01-17T17:03:29.040Z]     javax/jms/XATopicSession.class
```

**Comment by mareknovotny** [comment]
jenkins do fdb

**Comment by mareknovotny** [comment]
please be aware of Optaweb Vehicle rooting app failures which are very probably related to Spring framework upgrade. FYI @yurloc can help with assessing.

```
- org.optaweb.vehiclerouting.plugin.websocket.WebSocketErrorMessageSenderTest.should_send_consumed_message_over_websocket{SimpMessagingTemplate}
- org.optaweb.vehiclerouting.service.error.ErrorListenerTest.should_pass_error_message_to_consumer{ErrorMessageConsumer}
- org.optaweb.vehiclerouting.service.location.LocationServiceTest.createLocation{DistanceMatrixRow}
- org.optaweb.vehiclerouting.service.location.LocationServiceTest.addLocation{DistanceMatrixRow}

```

**Comment by yurloc** [comment]
Working on it...

**Comment by yurloc** [comment]
@mareknovotny @VaniHaripriya I have a fix (hopefully!) for `optaweb-vehicle-routing` (https://github.com/kiegroup/optaweb-vehicle-routing/pull/641) and `optaweb-employee-rostering` (https://github.com/kiegroup/optaweb-employee-rostering/pull/755).

Since build chain is unable to pick up feature branches from different forks, I've opened my own set of PRs to get a FDB result early but *I missed https://github.com/kiegroup/droolsjbpm-integration/pull/2680* so my attempt has failed.

I'm going to rerun FDB from my PR (https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1872), again, to get a result ASAP but feel free to kill it any time if you want to continue the work from this PR. Be aware that in that case @VaniHaripriya will need to clone optaweb updates on the `RHPAM-4111` branch from _my forks_ into _her forks_ for the FBD to pick up the changes if it starts from this PR.

**Pull request closed without merging**